ouroboros-consensus-test/test-consensus/Test/Consensus/MiniProtocol/BlockFetch/Client.hs

```diff
@@ -0,0 +1,424 @@
+            chainDbView
+
+        blockFetchCfg = BlockFetchConfiguration {
+              bfcMaxConcurrencyBulkSync = 3
+            , bfcMaxConcurrencyDeadline = 3
+            , bfcMaxRequestsInflight    = 10
+            , bfcDecisionLoopInterval   = 0.0
+            , bfcSalt                   = 0
+            }
+
+    _ <- forkLinkedThread registry "BlockFetchLogic" $
+      blockFetchLogic
+        nullTracer
+        nullTracer
+        blockFetchConsensusInterface
+        fetchClientRegistry
+        blockFetchCfg
+
+    let runBlockFetchClient peerId =
+          bracketFetchClient fetchClientRegistry ntnVersion peerId \clientCtx -> do
+            let bfClient = blockFetchClient
+                    ntnVersion
+                    (readTVar varControlMessage)
+                    nullTracer
+                    clientCtx
+                bfServer =
+                    blockFetchServerPeer $ mockBlockFetchServer getCurrentChain
+                  where
+                    getCurrentChain = atomically $ (Map.! peerId) <$> getCandidates
+
+                blockFetchTracer = Tracer \case
+                    (Driver.Client, ev) -> do
+                      atomically case ev of
+                        Driver.TraceRecvMsg (AnyMessageAndAgency _ (MsgBlock _)) ->
+                           modifyTVar varFetchedBlocks $ Map.adjust (+ 1) peerId
+                        _ -> pure ()
+                      traceWith tracer $
+                        show peerId <> ": BlockFetchClient: " <> show ev
+                    _ -> pure ()
+            fst <$> Driver.runConnectedPeersPipelined
+              createConnectedChannels
+              blockFetchTracer
+              codecBlockFetchId
+              bfClient
+              bfServer
+
+        -- On every tick, we schedule updates to the shared chain fragment
+        -- (mocking ChainSync).
+        forkTicking peerId =
+            forkLinkedWatcher registry ("TickWatcher " <> condense peerId) $
+              LogicalClock.tickWatcher clock \tick -> atomically do
+                let updates = toChainUpdates $
+                      Map.findWithDefault [] tick $
+                        getSchedule $ peerUpdates Map.! peerId
+                    updateChain chain =
+                      case Chain.applyChainUpdates updates chain of
+                        Just chain' -> chain'
+                        Nothing     -> error "Chain update failed"
+                -- Block until our "ChainSync" thread registered itself to the
+                -- FetchClientRegistry, see 'forkBracketSync' below.
+                _ <- blockUntilJust $ Map.lookup peerId <$> readTVar varChains
+                modifyTVar varChains $ Map.adjust updateChain peerId
+
+        forkKeepAlive peerId =
+          forkLinkedThread registry "KeepAlive" $
+            bracketKeepAliveClient fetchClientRegistry peerId \_ ->
+              infiniteDelay
+
+        forkBracketSync peerId =
```

Comment:
Similar comment as for forkKeepAlive.

Also, please call it "stubChainSync", not "BracketSync" to emphasize it's required just because the ChainSync mini protocol is assumed by the BlockFetch infra.

Lastly: since this and `forkTicking` both mock ChainSync, could we combine them into one ChainSync mock?

Reply:
Good idea, I will call `forkTicking` inside of `forkBracketSync` (instead of `infiniteDelay`) to make this more clear.